sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/variantExpressions.scala

```diff
@@ -39,27 +33,22 @@ import org.apache.spark.unsafe.types._
   group = "variant_funcs"
 )
 // scalastyle:on line.size.limit
-case class ParseJson(child: Expression) extends UnaryExpression
-  with NullIntolerant with ExpectsInputTypes with CodegenFallback {
+case class ParseJson(child: Expression)
+  extends UnaryExpression with RuntimeReplaceable with ImplicitCastInputTypes {
```

Comment:
let's not change the behavior in this PR. We should still extend `ExpectsInputTypes`

Reply:
Okay, let me update it.